sdk/identity/azure-identity/azure/identity/aio/_credentials/imds.py

```diff
@@ -40,34 +41,28 @@ async def _acquire_token_silently(self, *scopes: str, **kwargs: Any) -> Optional
         return self._client.get_cached_token(*scopes)
 
     async def _request_token(self, *scopes: str, **kwargs: Any) -> AccessToken:  # pylint:disable=unused-argument
-        if self._endpoint_available is None:
-            # Lacking another way to determine whether the IMDS endpoint is listening,
-            # we send a request it would immediately reject (because it lacks the Metadata header),
-            # setting a short timeout.
+
+        if within_credential_chain.get() and not self._endpoint_available:
```

Comment:
What about if self._endpoint_available is None: but if self._endpoint_available is None is false?

Reply:
The behavior here is that if `self._endpoint_available` is any falsey value (in our case, this would generally be `None`), then we perform the probe. This is to always allow probe retries with the same credential instance if it somehow failed before.